### PR TITLE
Added Network Threat Manager feature

### PR DESCRIPTION
**Pull request by pierantoniomerlino**
This PR adds the Network Threat Manager feature with APIs and their implementation.

**Related Issue:** This PR fixes/closes N/A

**Description of the solution adopted:** The Network Threat Manager is a feature for managing potential threats that comes from the network. A marker interface is added to identify all the bundles that belong to the Network Threat Manager. The web console groups the configurations of all the related bundles to a single tab under the Security section. 
The first implementation of the Network Threat Manager is the flooding protection that tries to prevent unwanted access to the device adding a set of rules to the firewall.

### CODE REVIEW

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/security/IntrusionDetectionConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
+package org.eclipse.kura.security;
```
Comment (MMaiero) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/security/IntrusionDetectionConfiguration.java]: Missing copyright header

File: kura/org.eclipse.kura.ids.manager/src/main/java/org/eclipse/kura/internal/floodingprotection/configuration/FloodingProtectionOptions.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
+package org.eclipse.kura.internal.floodingprotection.configuration;
```
Comment (MMaiero) [inline kura/org.eclipse.kura.ids.manager/src/main/java/org/eclipse/kura/internal/floodingprotection/configuration/FloodingProtectionOptions.java]: Missing copyright header

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtSecurityServiceImpl.java [hunk 3]
```diff
@@ -16,6 +16,7 @@
 import javax.servlet.http.HttpSession;
```
Comment (MMaiero) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtSecurityServiceImpl.java]: 2021?

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/service/GwtComponentService.java [hunk 4]
```diff
@@ -145,6 +145,27 @@
     public void updateComponentConfiguration(GwtXSRFToken xsrfToken, GwtConfigComponent configComponent)
             throws GwtKuraException;
 
+    /**
```
Comment (MMaiero) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/service/GwtComponentService.java:148]: 2021?

**Comment by mstankovic** [comment]
@pierantoniomerlino I have one issue here. can you please take a look:
my steps:
- fresh install this PR
- use eth0 to access UI
- go to firewall and edit port '443' to be opened for every interface (i just removed eth0), apply 
- try to access web UI over eth1 = successful 
- enable flooding protection
- disable flooding protection
- wait some time (about 1min)
- try to access webUi over eth1 - not able

NOTE: after that i need to edit 443 port again in order to access web ui with interface other then eth0

**Comment by mstankovic** [comment]
Hi, I tested this PR

#flooding protection:
- For kura it seems working fine. rules are applied to mangle table under the prerouting-kura chain. Verified this executing cmd: **sudo iptables -t mangle --list**

**Comment by pierantoniomerlino** [comment]
@mstankovic Thanks for the tests! I'll take a look at the errors in the logs.

File: kura/distrib/src/main/ant/build_equinox_distrib.xml [hunk 5]
```diff
@@ -178,6 +178,9 @@
 		<!-- Miscellaneous bundles config -->
 		<antcall target="misc-config" />
 
+		<!-- IDS bundles config -->
```
Comment (MMaiero) [inline kura/distrib/src/main/ant/build_equinox_distrib.xml]: Maybe we should remove ids also from here and write threat management

**Review by MMaiero — APPROVED** [review]

**Pull request merged**